src/controllers/csi/provisioner/controller.go

```diff
@@ -101,69 +101,102 @@ func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request r
+	// Create/update the dynakubeMetadata entry while `LatestVersion` is not necessarily set
 	// so the host oneagent-storages can be mounted before the standalone agent binaries are ready to be mounted
-	err = provisioner.createOrUpdateDynakube(oldDynakube, dynakube)
+	err = provisioner.createOrUpdateDynakubeMetadata(oldDynakubeMetadata, dynakubeMetadata)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	oldDynakube = *dynakube
+	oldDynakubeMetadata = *dynakubeMetadata
 
-	if err = provisioner.createCSIDirectories(provisioner.path.EnvDir(dynakube.TenantUUID)); err != nil {
-		log.Error(err, "error when creating csi directories", "path", provisioner.path.EnvDir(dynakube.TenantUUID))
+	if err = provisioner.createCSIDirectories(provisioner.path.EnvDir(dynakubeMetadata.TenantUUID)); err != nil {
+		log.Error(err, "error when creating csi directories", "path", provisioner.path.EnvDir(dynakubeMetadata.TenantUUID))
 		return reconcile.Result{}, errors.WithStack(err)
 	}
-	log.Info("csi directories exist", "path", provisioner.path.EnvDir(dynakube.TenantUUID))
+	log.Info("csi directories exist", "path", provisioner.path.EnvDir(dynakubeMetadata.TenantUUID))
 
-	latestProcessModuleConfig, storedHash, err := provisioner.getProcessModuleConfig(dtc, dynakube.TenantUUID)
+	latestProcessModuleConfigCache, requeue, err := provisioner.updateProcessModuleConfigCache(dtc, dynakubeMetadata, dk)
+	if requeue {
+		return reconcile.Result{RequeueAfter: defaultRequeueDuration}, err
+	}
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Set/Update the `LatestVersion` field in the database entry
+	err = provisioner.createOrUpdateDynakubeMetadata(oldDynakubeMetadata, dynakubeMetadata)
 	if err != nil {
-		log.Error(err, "error when getting the latest ruxitagentproc.conf")
 		return reconcile.Result{}, err
 	}
+
+	err = provisioner.writeProcessModuleConfigCache(dynakubeMetadata.TenantUUID, latestProcessModuleConfigCache)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{RequeueAfter: defaultRequeueDuration}, nil
+}
+
+func (provisioner *OneAgentProvisioner) updateProcessModuleConfigCache(dtc dtclient.Client, dynakubeMetadata *metadata.Dynakube, dk *dynatracev1beta1.DynaKube) (
+	latestProcessModuleConfigCache *processModuleConfigCache,
+	requeue bool,
+	err error,
+) {
+	latestProcessModuleConfig, storedHash, err := provisioner.getProcessModuleConfig(dtc, dynakubeMetadata.TenantUUID)
+	if err != nil {
+		log.Error(err, "error when getting the latest ruxitagentproc.conf")
+		return nil, false, err
+	}
 	latestProcessModuleConfig = latestProcessModuleConfig.AddHostGroup(dk.HostGroup())
-	latestProcessModuleConfigCache := newProcessModuleConfigCache(latestProcessModuleConfig)
+
+	connectionInfo, err := dtc.GetConnectionInfo()
+	if err != nil {
+		log.Error(err, "error when getting OneAgent connectionInfo")
+		return nil, false, err
+	}
+	latestProcessModuleConfig = latestProcessModuleConfig.AddConnectionInfo(connectionInfo)
```

Comment:
shouldn't there be a check if `codeModulesImage` is used ?  I think its only necessary to add the connection info stuff when image is used